src/3d/qgsvectorlayerchunkloader_p.h

```diff
@@ -0,0 +1,117 @@
+/***************************************************************************
+  qgsvectorlayerchunkloader_p.h
+  --------------------------------------
+  Date                 : July 2019
+  Copyright            : (C) 2019 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSVECTORLAYERCHUNKLOADER_P_H
+#define QGSVECTORLAYERCHUNKLOADER_P_H
+
+///@cond PRIVATE
+
+//
+//  W A R N I N G
+//  -------------
+//
+// This file is not part of the QGIS API.  It exists purely as an
+// implementation detail.  This header file may change from version to
+// version without notice, or even be removed.
+//
+
+#include "qgschunkloader_p.h"
+#include "qgsfeature3dhandler_p.h"
+#include "qgschunkedentity_p.h"
+
+class Qgs3DMapSettings;
+class QgsVectorLayer;
+class QgsVectorLayer3DTilingSettings;
+class QgsVectorLayerFeatureSource;
+class QgsAbstract3DSymbol;
+class QgsFeature3DHandler;
+
+#include <QFutureWatcher>
+
+
+/**
+ * \ingroup 3d
+ * This loader factory is responsible for creation of loaders for individual tiles
+ * of QgsVectorLayerChunkedEntity whenever a new tile is requested by the entity.
+ *
+ * \since QGIS 3.12
```

Comment:
maybe remove \ingroup and \since in private API?

Reply:
I don't have a strong opinion on this and there is no official policy about these - I would keep it as is...